skiko/src/jvmMain/cpp/windows/directx.cc

```diff
@@ -0,0 +1,70 @@
+#ifdef SK_DIRECT3D
+#include <stdexcept>
+#include <locale>
+#include <Windows.h>
+#include <jawt_md.h>
+#include "jni_helpers.h"
+
+#include "GrBackendSurface.h"
+#include "GrDirectContext.h"
+#include "SkSurface.h"
+
+#include "d3d/GrD3DTypes.h"
+#include <d3d12sdklayers.h>
+#include "d3d/GrD3DBackendContext.h"
+#include <d3d12.h>
+#include <dxgi1_4.h>
+#include <dxgi1_6.h>
+
+UINT adapterIndex = 0;
```

Comment:
I am not sure about this global variable, will it be thread-safe, or work with multiple SkiaLayer's

Can we get rid of it? It will much safer and cleaner. For example:

```
bool defineAdapter(IDXGIFactory4 *pFactory, IDXGIAdapter1 **ppAdapter, int* index)

Java_org_jetbrains_skiko_GraphicsApiKt_getNextDirectXAdapter(oldIndex: Int): Int
```